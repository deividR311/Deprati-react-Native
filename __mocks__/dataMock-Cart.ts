export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
