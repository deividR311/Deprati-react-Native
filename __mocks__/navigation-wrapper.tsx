import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
