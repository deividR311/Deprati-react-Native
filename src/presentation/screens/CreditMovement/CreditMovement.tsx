import { useRoute } from '@react-navigation/native';
import React from 'react';
import { View, Text, ScrollView } from 'react-native';
import { FontStyles } from '../../../application/common';
import { LocalStorageKey } from '../../../application/state-manager/services/localstorage';
import { useLocalStorage } from '../../../application/state-manager/services/localstorage/useLocalStorage';
import { formatToCurrecy } from '../../../application/utils/currency';
import { useEmmaSdk } from '../../../infrastucture/native-modules/emma';
import CardReload from '../../common-components/cardReload/cardReload';
import { CreditInformationChart } from '../../common-components/credit-information-chart';
import { SafeArea } from '../../common-components/safe-area';
import { styles } from './CreditMovement.stylesheet';
import { useCreditMovement } from './hook/useCreditMovement.hook';
import { SkeletonPage } from './skeleton';
import NextPayment from '../Credit/components/NextPayment';
import { PopupWhatsapp } from '../../common-components/poppup/poppupWhatsapp';

const CreditMovement = () => {
  const route = useRoute();
  useEmmaSdk({ route });
  const {
    loading,
    hasError,
    contentPayment,
    constentChart,
    contentMovement,
    getDataContent,
    showModalError,
    showModalTextError,
    closeModaError
  } = useCreditMovement();
  const {
    localStorageData: {
      [LocalStorageKey.AccountDisplayNumber]: AccountDisplayNumber,
      [LocalStorageKey.AccountAdditionalNumber]: ADDITIONAL_ACCOUNT_NUMBER
    }
  } = useLocalStorage();

  const LastMovement = () => {
    return (
      <View style={styles.lastMovement}>
        <Text style={styles.lastMovement_title}>Últimos movimientos</Text>
        <View style={styles.lastMovement_movement}>
          {contentMovement.map((item, index: number) => {
            return (
              <LastMovementItem key={`lastmovement${index}`} item={item} />
            );
          })}
        </View>
      </View>
    );
  };

  const LastMovementItem = ({ item }) => {
    let typeMovement = true;
    const typesMovement = {
      COMPRA: false,
      DEVOLUCIÓN: true,
      PAGO: true
    };
    typeMovement = typesMovement[item?.tipoMovimiento] ?? true;

    return (
      <View style={styles.lastMovementItem}>
        <Text style={styles.lastMovementItem__title}>
          {item?.descripcionMovimiento}
        </Text>
        <View style={styles.lastMovementItem__data}>
          <Text style={styles.lastMovementItem__data_date}>
            {item?.fechaMovimiento}
          </Text>
          <Text
            style={[
              styles.lastMovementItem__data_price,
              typeMovement ? styles.price_plus : styles.price_minus
            ]}>
            {typeMovement ? '+' : '-'}{' '}
            {formatToCurrecy(item?.valor?.toString())}
          </Text>
        </View>
      </View>
    );
  };

  if (loading) return <SkeletonPage />;

  return (
    <>
      {showModalTextError === '' ? (
        <>
          <SafeArea>
            <ScrollView
              showsVerticalScrollIndicator={false}
              contentContainerStyle={styles.contentContainer}>
              <CardReload
                style={{ paddingVertical: 8 }}
                typeCard={'information'}
                text={
                  <Text style={FontStyles.Body_2}>
                    Ha ocurrido un error de carga,&nbsp;
                    <Text style={[FontStyles.Bold]}>
                      vuelve a cargar esta sección aquí.
                    </Text>
                  </Text>
                }
                isVisible={hasError}
                onClose={() => {
                  getDataContent();
                }}
                onPress={args => {
                  getDataContent();
                }}
              />
              <View style={styles.content_chart}>
                <CreditInformationChart
                  constentChart={constentChart}
                  totalAmount={Number(
                    constentChart?.totalAmount
                      ?.toString()
                      .replace('.', '')
                      .replace(',', '.')
                  )}
                  amountSpent={Number(
                    constentChart?.amountSpent
                      .toString()
                      .replace('.', '')
                      .replace(',', '.')
                  )}
                  ownerFullname={constentChart?.clientName}
                  affiliateDate={constentChart?.affiliateDate}
                  cardNumber={constentChart?.cardNumber}
                  availableAmount={Number(
                    constentChart?.availableAmount
                      .toString()
                      .replace('.', '')
                      .replace(',', '.')
                  )}
                  numeroTarjetaAdicionalDisplay={AccountDisplayNumber}
                  showAffiliateDate={false}
                  showDisplayCard={false}
                />
              </View>

              <NextPayment
                contentPayment={contentPayment}
                loading={false}
                stylesCustom={{
                  nextPayment: styles.nextPayment__container
                }}
              />
              {ADDITIONAL_ACCOUNT_NUMBER === '00' && <LastMovement />}
            </ScrollView>
          </SafeArea>
        </>
      ) : (
        <>
          <PopupWhatsapp
            visible={showModalError}
            title={showModalTextError}
            onClose={closeModaError}
          />
        </>
      )}
    </>
  );
};

export default CreditMovement;
