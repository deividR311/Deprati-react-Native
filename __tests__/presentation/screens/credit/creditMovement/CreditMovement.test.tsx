import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../../__mocks__/navigation-wrapper';
import CreditMovement from '../../../../../src/presentation/screens/CreditMovement/CreditMovement';

let screenTest: any;
describe('renders CreditMovement', () => {
  it('renders correctly in Snapshot', () => {
    screenTest = render(<NavigationWrapper screen={CreditMovement} />);
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
