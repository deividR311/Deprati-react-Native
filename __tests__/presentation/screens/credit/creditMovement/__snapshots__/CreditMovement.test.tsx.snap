// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CreditMovement renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "flexDirection": "column",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "marginVertical": 8,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#E1E9EE",
                  "borderRadius": 4,
                  "height": 200,
                  "key": "text-1",
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "height": "100%",
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "translateX": undefined,
                        },
                      ],
                    },
                  ]
                }
              >
                <BVLinearGradient
                  colors={
                    Array [
                      4292995566,
                      4294113532,
                      4292995566,
                    ]
                  }
                  endPoint={
                    Object {
                      "x": 1,
                      "y": 0,
                    }
                  }
                  locations={null}
                  startPoint={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Object {
                "marginVertical": 8,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#E1E9EE",
                  "borderRadius": 4,
                  "height": 100,
                  "key": "text-1",
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "height": "100%",
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "translateX": undefined,
                        },
                      ],
                    },
                  ]
                }
              >
                <BVLinearGradient
                  colors={
                    Array [
                      4292995566,
                      4294113532,
                      4292995566,
                    ]
                  }
                  endPoint={
                    Object {
                      "x": 1,
                      "y": 0,
                    }
                  }
                  locations={null}
                  startPoint={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "width": "100%",
              }
            }
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "paddingHorizontal": 2,
                  "paddingVertical": 2,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 60,
                    "key": "cateogry-1",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
